src/command/vanilla/TpCommand.ts

```diff
@@ -1,121 +1,115 @@
-                    if (!target) {
-                        sender.sendMessage(`§c${args[0]} is not online!`);
-                        return;
+                sender.sendMessage(`You have been teleported to ${args[0]}, ${args[1]}, ${args[2]}`);
+                sender.sendMessage(`Teleported ${sender.name} to ${args[0]}, ${args[1]}, ${args[2]}`);
+
+            } else {
+                sender.sendMessage('§cSyntax error: Invalid position!');
+            }
+        } else if (typeof args[0] == 'string') {
+            // /tp <player> -> <x> <y> <z>
+
+            if (typeof args[1] == 'number') {
+
+                if (typeof args[2] == 'number' && typeof args[3] == 'number') {
+                    let target = sender.getServer().getPlayerByName(args[0]);
+    
+                    if (target) {
+    
+                        target.x = args[1];
+                        target.y = args[2];
+                        target.z = args[3];
+    
+                        target.broadcastMove(target, MovementType.Teleport);
+    
+                        target.sendMessage(`You have been teleported to ${args[1]}, ${args[2]}, ${args[3]}`);
+                        sender.sendMessage(`Teleported ${target.name} to ${args[1]}, ${args[2]}, ${args[3]}`);
+    
+                    } else {
+                        sender.sendMessage('§cNo targets matched selector!');
                     }
+                } else {
+                    sender.sendMessage('§cSyntax error: Invalid position!');
+                }
+
+            } else if (typeof args[1] == 'string') {
+                // /tp <player> -> <player>
+
+                let targetFrom = sender.getServer().getPlayerByName(args[0]);
+                let targetTo = sender.getServer().getPlayerByName(args[0]);
+
+                if (targetFrom && targetTo) {
+
+                    targetFrom.x = targetTo.x;
+                    targetFrom.y = targetTo.y;
+                    targetFrom.z = targetTo.z;
+
+                    targetFrom.broadcastMove(targetFrom, MovementType.Teleport);
+
+                    targetFrom.sendMessage(`You have been teleported to ${targetTo.name}`);
+                    sender.sendMessage(`Teleported ${targetFrom.name} to ${targetTo.name}`);
 
-                    player.x = target.x;
-                    player.y = target.y;
-                    player.z = target.z;
                 } else {
-                    player.y = args[1];
+                    sender.sendMessage('§cNo targets matched selector!');
                 }
-                break;
-            case 3:
-
-                player.x = args[1];
-                player.z = args[2];
-                break;
-            case 4:
-                player.x = args[1];
-                player.y = args[2];
-                player.z = args[3];
-                break;
-        }
 
-        player.broadcastMove(player);
-        sender.sendMessage(`Teleported ${args[0]} to ${args.slice(1).join(' ')}`);
```

Comment:
This automatically handles the output of where you've been teleported instead of using duplicated code.